gramps/gen/filters/_genericfilter.py

```diff
@@ -282,9 +278,37 @@ def apply(self, db, id_list=None, tupleind=None, user=None, tree=False):
             raise Exception("invalid operator: %r" % self.logical_op)
 
         start_time = time.time()
-        res = self.apply_logical_op_to_all(
-            db, id_list, apply_logical_op, user, tupleind, tree
-        )
+
+        # build the starting set of possible_handles to be filtered
+        possible_handles: Set[PrimaryObjectHandle]
+        if id_list is not None:
+            if tupleind is not None:
```

Comment:
```suggestion
        if id_list:
            if tupleind is not None:
```
Good catch on tupleind to avoid problems when it is zero. I'm not sure we need to use `is not None` for id_list though